### PR TITLE
Issue_1943: get rid of ticket_key_name tag from ssl_multicert

### PR DESCRIPTION
**Pull request by persiaAziz**
ticket_key_name is removed. 
Ticket key should be set in records.config like 
CONFIG proxy.config.ssl.server.ticket_key.filename STRING ssl_ticket.key

### CODE REVIEW

**Comment by zwoop** [comment]
Ah cool, this is to fix @mlibbey's problem ? Please update the labels, milestone (7.1.0) etc. :-)

**Comment by persiaAziz** [comment]
yes it does. ticket_key_name is being by overridden by default keyblock. Hence removing it

File: iocore/net/SSLUtils.cc [hunk 1]
```diff
@@ -1939,10 +1933,6 @@ ssl_extract_certificate(const matcher_line *line_info, ssl_user_config &sslMultC
       sslMultCertSettings.session_ticket_enabled = atoi(value);
     }
 
-    if (strcasecmp(label, SSL_SESSION_TICKET_KEY_FILE_TAG) == 0) {
```
Comment (masaori335) [inline iocore/net/SSLUtils.cc]: Should definition of SSL_SESSION_TICKET_KEY_FILE_TAG also removed?

Comment (persiaAziz) [inline iocore/net/SSLUtils.cc]: good catch! .. yes it should

**Comment by masaori335** [comment]
The section of `ticket_key_name` in docs of ssl_multicert.config should be also removed.
https://docs.trafficserver.apache.org/en/latest/admin-guide/files/ssl_multicert.config.en.html?highlight=ticket_key_name

**Comment by persiaAziz** [comment]
yes I will update the doc to say that this has been removed and keyblock should be set in a different way

**Comment by masaori335** [comment]
Great! I think TLS Session Tickets is widely used. Because it's cheap but effective.

**Comment by zwoop** [comment]
Do we need this PR? Is this sufficiently different from what's on master right now? If not, I'd say just close this, and focus reviews / development on the master branch PR first.

**Comment by persiaAziz** [comment]
closing this one. tracking in the PR for  master

File: doc/admin-guide/files/ssl_multicert.config.en.rst [hunk 2]
```diff
@@ -98,20 +98,8 @@ ssl_ticket_enabled=1|0 (optional)
...
-  This option has no effect if session tickets are disabled by the
-  ``ssl_ticket_enabled`` option.  The contents of the key file should
-  be 48 random (ASCII) bytes. One way to generate this would be to run
-  ``head -c48 /dev/urandom | openssl enc -base64 | head -c48 > file.ticket``.
-
-  Session ticket support is enabled by default. If neither of the
-  ``ssl_ticket_enabled`` and ``ticket_key_name`` options are
-  specified, and internal session ticket key is generated. This
-  key will be different each time Traffic Server is started.
+ticket_key_name=FILENAME (optional) [**REMOVED in 7.1.x and 8.0**]
```
Comment (mlibbey) [inline doc/admin-guide/files/ssl_multicert.config.en.rst]: And irrelevent in 7.0.

**Comment by persiaAziz** [comment]
@mlibbey Do you mean if ATS can see the changes in content of the ticket file ? If I change the content of the file without changing the filename, then `traffic_ctl reload` is not going to reload the ticket key. But if the name of the file changes in records.config, then ATS reloads the content since the config itself is reloadable.

**Comment by mlibbey** [comment]
proxy.config.ssl.server.multicert.filename contains the name and location of ssl_multicert.config. That file is reloaded during a config reload (I think as long as something --remap/records/or ssl_multicert is changed?) I'd think that proxy.config.ssl.server.ticket_key.filename  should similarly get reloaded so that operators don't have to update records.config with a new filename for every key rotation.

**Comment by jpeach** [comment]
This was implemented for SSL certificates, so you can do the same thing for the ticket key.

**Pull request closed without merging**